frontend/discourse/app/components/create-topic-button.gjs

```diff
@@ -1,23 +1,40 @@
 import Component from "@glimmer/component";
 import { tracked } from "@glimmer/tracking";
+import { service } from "@ember/service";
 import DButton from "discourse/components/d-button";
 import TopicDraftsDropdown from "discourse/components/topic-drafts-dropdown";
 import concatClass from "discourse/helpers/concat-class";
+import { applyValueTransformer } from "discourse/lib/transformer";
 
 export default class CreateTopicButton extends Component {
+  @service router;
+  @service siteSettings;
+  @service currentUser;
+
   @tracked btnTypeClass = this.args.btnTypeClass || "btn-default";
   @tracked label = this.args.label ?? "topic.create";
 
   btnId = this.args.btnId ?? "create-topic";
 
+  get permissionClass() {
```

Comment:
It would be better if you merged with `btnTypeClass` and use a single getter for both